### DISCUSSION TITLE
URL Decoding not working correctly, e.g. Eastern European characters not correctly parsed

### DISCUSSION THREAD

**Discussion by PranshuBansalDev** [fastapi/fastapi#10513] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


@app.get("/{country}/{region}")
def get_items(country: str, region: str):
	return region
```

### Description

1. Open postman
2. Use this get request URL: http://localhost:7071/api/plants/Poland/%C5%9Awi%C4%99tokrzyskie
3. Notice that the value of `{region}` is incorrect when received at params portion

`Å\x9awiÄ\x99tokrzyskie`

Should be

`Świętokrzyskie`

### Operating System

Windows

### FastAPI Version

0.104.0

### Pydantic Version

2.4.2

### Python Version

Python 3.10.11

### Additional Context

It seems like its very difficult to get the actual raw string sent over the wire for the path parameters, am I missing something trivial here to get at that?

Even accessing `request` seems to give me the wrong encoding

I'd be happy to receive the string with the % in it as so: `%C5%9Awi%C4%99tokrzyskie`

**Comment by jgould22** [comment, marked as answer]
Postman is likely trying to "helpfully" guess that you wanted that url encoded and attempts to encode it for you

```bash
curl  http://localhost:8000/Poland/%C5%9Awi%C4%99tokrzyskie
"Świętokrzyskie"
```

```bash
INFO:     127.0.0.1:56450 - "GET /Poland/%C5%9Awi%C4%99tokrzyskie HTTP/1.1" 200 OK
```
The solution is don't use postman in this case or find out if it can be disabled in the settings.

edit: here https://learning.postman.com/docs/sending-requests/requests/#choosing-custom-settings

**Comment by Tubaher** [comment]
Hello! Well, if you need a fast solution this is the code that works for me to get the exact `region` parameter: 

 ```python
import urllib.parse

@app.get("/{country}/{region}")
def get_items(country: str, region: str):
    encoded_region = urllib.parse.quote(region, safe="")
    return encoded_region
```

The explanation is that the URL path parameters are automatically decoded from URL encoding (percent-encoding). Since the region parameter contains % characters, FastAPI will automatically decode these percent-encoded characters. Consequently, you see "Świętokrzyskie", which is properly formatted in UTF-8. The provided code example re-encodes the region parameter, performing the inverse operation to obtain the original encoded string "%C5%9Awi%C4%99tokrzyskie".
